js-watcher/server.go

```diff
@@ -0,0 +1,61 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+
+	"github.com/julienschmidt/httprouter"
+)
+
+type Server struct {
+	caches CacheCollection
+	config Config
+	router *httprouter.Router
+	server *http.Server
+}
+
+type LoggingResponseWriter struct {
+	http.ResponseWriter
+	req http.Request
+}
+
+func (r LoggingResponseWriter) WriteHeader(status int) {
```

Comment:
Did you think of using a middleware for this such as https://echo.labstack.com/middleware/

Reply:
Yeah if we switch to a more extensive server library that is easy. But you can make middlewares even with the built-in http server library. The problem is that middlewares for `net/http` take a http [HandlerFunc](https://pkg.go.dev/net/http#HandlerFunc) and return another one. But in the http HandlerFunc we only get a [`ResponseWriter` interface](https://pkg.go.dev/net/http#ResponseWriter) and this does not allow me to read the response or status code. Only write them (in some unreachable location). That is why I had to wrap the response writer like this so that I can access the status code.

Reply:
For now I dont want to re-write this whole thing with echo or gin just for the sake of better logging. I can open a PR for that as an improvement down the road.